### DISCUSSION TITLE
Create 3 robot.say does nothing

### DISCUSSION THREAD

**Discussion by switchdoclabs** [iRobotEducation/create3_docs#147] [Questions] [answered]
Hi,

I'm running the Python Playground example programs and the programs that use "robot.say" are not saying anything.  I can play notes, so I know the speaker is working.

Name: iRobot-05F8BBBCD
Battery:  16601 mV;  100 %
Serial #: JALH21032R
SKU: RCi3099

Any suggestions?  Running the latest firmware version.

John Shovic

**Comment by shamlian** [comment, marked as answer]
"Say phrase" is not currently implemented on the Create 3 robot (it is, however, implemented on the Root robot). Sorry for the inconvenience; you can play single notes through the Python interface or compose longer sequences using [ROS 2](https://iroboteducation.github.io/create3_docs/examples/actuators-cli/#audio).

**Reply by shamlian** [reply]
I just double-checked, and all examples that use "say" are in the "root" folder tree, and there are no instances of "say" in the "tutorial" or "create3" code. Please do be sure you are exercising code which is intended for the Create 3 robot.

**Reply by switchdoclabs** [reply]
Thank you!  You are correct.  All of the examples are in the "root" tree.

**Comment by jerrynsr** [comment]
If you want to add audio to the Create3 Robot and you have a RPI4 or Jetson board connected you can get one of these:
https://www.amazon.com/gp/product/B075M7FHM1/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
The speaker fits fine with my RPI4 and Addata SSD in the cargo bay.  I just ordered a micro 4 port hub and will be extending the GPIO lines up to a second deck.  I am working on the stl for the second deck.  
Here's my Create3 roaming and barking like a dog:
https://youtu.be/b0KGXZB1iyQ
I also have the ros2 text to speech working.
